lua/astrocommunity/pack/astro/init.lua

```diff
@@ -17,4 +39,33 @@ return {
     "jay-babu/mason-nvim-dap.nvim",
     opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "js") end,
   },
+  {
+    "jay-babu/mason-null-ls.nvim",
+    opts = function(_, opts)
+      opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, { "eslint_d" })
+      if not opts.handlers then opts.handlers = {} end
+
+      local has_eslint = function(util)
+        return util.root_has_file ".eslintrc.js"
+          or util.root_has_file ".eslintrc.cjs"
+          or util.root_has_file ".eslintrc.yaml"
+          or util.root_has_file ".eslintrc.yml"
+          or util.root_has_file ".eslintrc.json"
+          or check_json_key_exists(vim.fn.getcwd() .. "/package.json", "eslintConfig")
+      end
+
+      opts.handlers.eslint_d = function()
+        local null_ls = require "null-ls"
+        null_ls.register(
+          null_ls.builtins.diagnostics.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
+        )
+        null_ls.register(
+          null_ls.builtins.formatting.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
+        )
+        null_ls.register(
+          null_ls.builtins.code_actions.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
+        )
```

Comment:
```suggestion
        local null_ls = require "null-ls"
        if null_ls.builtins.diagnostics.eslint_d then
        	utils.list_insert_unique(
	        null_ls.builtins.diagnostics.eslint_d.filetypes , "astro")
	else 
	null_ls.register(
          null_ls.builtins.diagnostics.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
        )
        end
        if null_ls.builtins.code_actions.eslint_d then
        	utils.list_insert_unique(
        	null_ls.builtins.code_actions.eslint_d.filetypes , "astro")
        else
        	null_ls.register(
          null_ls.builtins.formatting.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
        )
        if null_ls.builtins.formatting.eslint_d then 
        
        utils.list_insert_unique(
        null_ls.builtins.formatting.eslint_d.filetypes , "astro")
        else
        null_ls.register(
          null_ls.builtins.code_actions.eslint_d.with { condition = has_eslint, extra_filetypes = "astro" }
        )
        end
        
```

Reply:
@rafaelderolez  I think this should work? 🤞🏾

Reply:
I'll commit your suggestion and give it a whirl 🫡

Reply:
@Uzaaft no that breaks eslint_d alltogether!